File: backend/src/main/java/com/back/domain/chat/entity/ChatMessage.java
package com.back.domain.chat.entity;

import com.back.domain.member.entity.Member;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.Lob;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Entity
@Getter
@Table(name = "chat_message")
@EntityListeners(AuditingEntityListener.class)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ChatMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "chat_message_id")
    private Long id;

    @Lob
    @Column(name = "chat_message_content", nullable = false)
    private String content;

    @CreatedDate
    private LocalDateTime sentAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chat_room_id")
    private ChatRoom chatRoom;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sender_id")
    private Member sender;


    @Builder
    public ChatMessage(String content, ChatRoom chatRoom, Member sender) {
        this.content = content;
        this.chatRoom = chatRoom;
        this.sender = sender;
        this.sentAt = LocalDateTime.now();
    }
}

File: backend/src/main/java/com/back/domain/chat/entity/ChatMessage.kt
package com.back.domain.chat.entity

import com.back.domain.member.entity.Member
import jakarta.persistence.*
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@Entity
@Table(name = "chat_message")
@EntityListeners(AuditingEntityListener::class)
class ChatMessage private constructor(
    @field:Column(name = "chat_message_content", nullable = false)
    @field:Lob
    var content: String?,
    
    @field:ManyToOne(fetch = FetchType.LAZY)
    @field:JoinColumn(name = "chat_room_id")
    var chatRoom: ChatRoom,
    
    @field:ManyToOne(fetch = FetchType.LAZY)
    @field:JoinColumn(name = "sender_id")
    var sender: Member
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "chat_message_id")
    var id: Long? = null
        private set

    @CreatedDate
    @Column(updatable = false)
    var sentAt: LocalDateTime? = LocalDateTime.now()
        private set

    companion object {
        fun create(
            content: String?,
            chatRoom: ChatRoom,
            sender: Member
        ): ChatMessage {
            return ChatMessage(
                content = content,
                chatRoom = chatRoom,
                sender = sender
            )
        }
    }
}
